docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -242,6 +242,47 @@ If you need to render the unescaped value:
 <1> `title` that resolves to `Expressions & Escapes` will be rendered as `Expressions &amp;amp; Escapes`
 <2> `paragraph` that resolves to `<p>My text!</p>` will be rendered as `<p>My text!</p>`
 
+===== Virtual Methods
+
+A virtual method is a part of an expression that looks like a regular Java method invocation.
+It's called "virtual" because it does not have to match the actual method of a Java class.
```

Comment:
How different is a virtual method from a `@TemplateExtension`?

Reply:
Virtual method is just a part of an expression. `@TemplateExtension` is one of the ways how to evaluate a virtual method. Others include the value resolvers generated for `@TemplateData` or classes used in parameter declarations - in this case a virtual method invocation is delegated to an existing instance method. But in fact, you can register a custom value resolver like in [VirtualMethodParametersTest](https://github.com/quarkusio/quarkus/pull/7707/files#diff-f0acf76ec898f234f2f810b6f7d5a437R13-R36) that is not backed by any Java method.

Reply:
It would be nice to add a section explaining that